src/main/common/pwl.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+
+#include "pwl.h"
+
+
+void pwlInitialize(pwl_t *pwl, float *yValues, int numPoints, float xMin, float xMax) {
+    pwl->numPoints = numPoints;
+    pwl->xMin = xMin;
+    pwl->xMax = xMax;
+    pwl->dx = (xMax - xMin) / (numPoints - 1);
+    pwl->yValues = yValues;
+}
+
+void pwlFill(pwl_t *pwl, float (*function)(float))
```

Comment:
Allow passing (opaque) argument to function
```suggestion
void pwlFill(pwl_t *pwl, float (*function)(float, void*), void* arg)
```